paper.tex

```diff
@@ -228,33 +250,127 @@ \subsection*{High-level model specification in YAML}
+\subsection*{Unambiguous low-level representation}
 
-    \item Drawing parameters from distributions. Why we are static.
-    % https://github.com/popsim-consortium/demes-paper/issues/3
+\jkcomment{Not sure if there's much point in this as a separate section any
+more? This could really be said in the previous section.}
+
+Parsers output the low-level description of Demes models,
+which is an explicit representation
+of the model where all parameters have been assigned values, and
+all data constraints have been checked. The output is formally defined
+as JSON, but in practise parsers output an object model that
+is suitable for the particular programming language, and
+that can be used directly by the implementing program.
+
+
+\subsection*{Scope of the specification}
+\jkcomment{This section is very rough.}
+
+% https://github.com/popsim-consortium/demes-paper/issues/3
+
+The Demes specification is by design simple and static---we wish to
+unambiguously describe a demographic model with fully specified
+and fixed parameters. This simplicity means that we cannot directly
+specify distributions or estimated confidence intervals
+for those parameters. While it is not difficult to imagine extending
+the specification in ways that would allows us to do this, it would
+greatly increase the complexity of parsers and would likely be of
+little practical use in any case. This is because the parameters of
+demographic models are typically tightly coupled, and cases in which
+distributions for different parameters can be simply described are rare.
+In this situation, the simplest way to describe estimated
+distribution is simply to list a large number samples from
+the posterior. While writing out a large number of Demes models in
+YAML format may seem inefficient, it can in fact be a compact
+way to describe these distributions.
+\jkcomment{TODO: add in a quick example here where we take an IM model
+and simulate a posterior distribution under some reasonable model. Report
+how much space it takes to represent a large number of samples from this
+distribution}
+
+Similarly, one might be interested in running simulations in which
+the demographic model parameters are drawn from a distribution, e.g.,
+in ABC inference~\citep{beaumont2002approximate}. Allowing for
+the many distributions of interest would greatly increase the complexity
+parsers, with a relatively limited benefit. It would be far simpler
+and more powerful to write some code to sample directly from the
+distributions of interest, using, for example the Demes Python
+API.
+\jkcomment{This is garbled, but we get the point? You're probably never
+going to be doing ABC from a completely general Demes model as input.
+There will be some basic model that you have in mind, that you're generating
+distributions of parameters around. So, what's the point in writing
+general machinery to do this sampling, when you could write it yourself in
+10 lines of Python?}
```

Comment:
Yep. One could even output a large sample of models from one's prior distribution in this way, then feed the prior models into a downstream simulator which then doesn't need to know anything about what you're doing. The disadvantage is that one can't neatly express the prior distribution of models as a single YAML file and attach that file to a publication.

Reply:
Right - we need to get across the point that this is in principle impossible though. If there was a concise way of writing down the joint distribution of all the parameters, we wouldn't be doing simulations in the first place.